### PR TITLE
Lint Print Statement 

### PR DESCRIPTION
**Pull request by noklam**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

[I am guilty of this](https://github.com/kedro-org/kedro/pull/3013/files#r1325813448), so this is for safeguarding myself 😅
At first found [this](https://pypi.org/project/remove-print-statements/), then I found `ruff` already handle this, perfect!  https://beta.ruff.rs/docs/rules/print/

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by merelcht — APPROVED** [review]
Nice find! 👍

**Review by stichbury — APPROVED** [review]
LGTM 🥇

File: kedro/templates/project/{{ cookiecutter.repo_name }}/src/requirements.txt [hunk 1]
```diff
@@ -12,4 +12,5 @@ nbstripout~=0.4
 pytest-cov~=3.0
 pytest-mock>=1.7.1, <2.0
 pytest~=7.2
+# Pin problematic traitlets release -  https://github.com/kedro-org/kedro/pull/3040
```
Comment (astrojuanlu) [inline kedro/templates/project/{{ cookiecutter.repo_name }}/src/requirements.txt:15]: nit, optional:

```suggestion
# Pin problematic traitlets release -  https://github.com/jupyter/notebook/issues/7048
```

**Comment by noklam** [comment]
Have to fix the install a bit, as the e2e test is not using pip install directly so it treats comment as actual requirments.

**Pull request merged**